quinn/src/endpoint.rs

```diff
@@ -349,17 +383,8 @@ impl EndpointInner {
             });
         let mut iovs = unsafe { iovs.assume_init() };
         loop {
-            match self.socket.poll_recv_ready(cx) {
```

Comment:
It feels weird that this code got added in the previous commit and is removed here. How does the intermediate state make sense?

Reply:
The logic is initially moved out of quinn-udp to make that crate async-agnostic in the previous commit, then pushed down into the runtime abstraction in this commit. I think decoupling those steps is a net win to separate those two concerns, even though it requires this code to move around a bit.